controllers/auth.js

```diff
@@ -86,10 +86,8 @@ const storeUserDeviceInfo = async (req, res) => {
 
     const userInfo = await QrCodeAuthModel.storeUserDeviceInfo(userJson);
 
-    if (!userInfo) {
-      return res.status(404).json({
-        message: BAD_REQUEST,
-      });
+    if (userInfo.userExists !== undefined && !userInfo.userExists) {
+      return res.boom.notFound("Document not found!");
```

Comment:
why do we need an exclusive check for undefined !
```suggestion
 if ( !userInfo?.userExists) {
```